compiler-core/src/type_/tests.rs

```diff
@@ -1426,6 +1512,23 @@ fn unused_variable_warnings_test() {
     assert_no_warnings!("pub fn a(b) { case b { #(c, _) -> c } }");
 }
 
+#[test]
+fn unused_imported_module_warnings_test() {
+    assert_import_warning!(
+        "import gleam/foo",
+        Warning::UnusedImportedModule {
+            name: "foo".to_string(),
+            location: SrcSpan { start: 7, end: 16 },
+        }
+    );
+
+    // Using imported function
+    assert_no_import_warning!("import gleam/foo; pub fn bar() { foo.foo() }");
+
+    // Using imported type
+    assert_no_import_warning!("import gleam/foo; pub fn bar(a: foo.Baz) { a }");
+}
```

Comment:
Can we have 1 assertion per test? We're slowly migrating all the tests to that.

Could we get a test that no warnings are emitted if an imported module isn't used but does import unqualified values